src/OpenTelemetry.Api/Trace/IActivityEnumerator.cs

```diff
@@ -14,21 +14,21 @@
 // limitations under the License.
 // </copyright>
 
-using System.Collections.Generic;
 using System.Diagnostics;
 
 namespace OpenTelemetry.Trace
 {
     /// <summary>
-    /// An interface used to perform zero-allocation enumeration of <see cref="Activity"/> tags. Implementation must be a struct.
+    /// An interface used to perform zero-allocation enumeration of <see cref="Activity"/> elements. Implementation must be a struct.
     /// </summary>
-    public interface IActivityTagEnumerator
+    /// <typeparam name="T">Enumerated item type.</typeparam>
+    public interface IActivityEnumerator<T>
```

Comment:
I switched `IActivityTagEnumerator` to `IActivityEnumerator<T>` so I could use it for tags, links, and events (up next). Breaking change, which is a con, but pro is with this change we don't need a bunch of interfaces that are basically the same, just different item types.